src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Debug/DebugProfileDebugTargetGenerator.cs

```diff
@@ -60,23 +60,23 @@ public override IReceivableSourceBlock<IProjectVersionedValue<IReadOnlyList<IEnu
             }
         }
 
-        private ILaunchSettingsProvider LaunchSettingProvider { get; }
+        private IVersionedLaunchSettingsProvider LaunchSettingProvider { get; }
         private IProjectThreadingService ProjectThreadingService { get; }
 
         /// <summary>
         /// This provides access to the class which creates the list of debugger values..
         /// </summary>
         public Task<IDynamicEnumValuesGenerator> GetProviderAsync(IList<NameValuePair>? options)
             => Task.FromResult<IDynamicEnumValuesGenerator>(
-                new DebugProfileEnumValuesGenerator(LaunchSettingProvider, ProjectThreadingService));
+                new DebugProfileEnumValuesGenerator((ILaunchSettingsProvider)LaunchSettingProvider, ProjectThreadingService));
 
         protected override void Initialize()
         {
-            IPropagatorBlock<ILaunchSettings, IProjectVersionedValue<IReadOnlyList<IEnumValue>>> debugProfilesBlock = DataflowBlockSlim.CreateTransformBlock<ILaunchSettings, IProjectVersionedValue<IReadOnlyList<IEnumValue>>>(
+            IPropagatorBlock<IProjectVersionedValue<ILaunchSettings>, IProjectVersionedValue<IReadOnlyList<IEnumValue>>> debugProfilesBlock = DataflowBlockSlim.CreateTransformBlock<IProjectVersionedValue<ILaunchSettings>, IProjectVersionedValue<IReadOnlyList<IEnumValue>>>(
                 update =>
                 {
                     // Compute the new enum values from the profile provider
-                    var generatedResult = DebugProfileEnumValuesGenerator.GetEnumeratorEnumValues(update).ToImmutableList();
+                    var generatedResult = DebugProfileEnumValuesGenerator.GetEnumeratorEnumValues(update.Value).ToImmutableList();
```

Comment:
Unrelated to this PR, but `ToImmutableArray` might be better here. It has a compact (array-based) representation in memory, which is better if the consumer is not intending to create mutated forms. Given it's exposed as `IReadOnlyList` I think that's a fair assumption in this case.